src/autocomplete.mjs

```diff
@@ -108,7 +113,9 @@ export default class extends Controller {
           )
           if (selected && !this.resultsTarget.hidden) {
             this.commit(selected)
-            event.preventDefault()
+            if (!this.submitOnEnterValue){
```

Comment:
Check out https://stimulus.hotwire.dev/reference/values#properties-and-attributes

Give `this.hasSubmitOnEnterValue` a try.